Sources/NIO/SocketChannel.swift

```diff
@@ -122,7 +122,7 @@ final class SocketChannel: BaseSocketChannel<Socket> {
         var result = ReadResult.none
         for i in 1...maxMessagesPerRead {
             guard self.isOpen && !self.inputShutdown else {
-                return result
+                throw ChannelError.eof
```

Comment:
This is a surprising behavioural change: why was it made?

Reply:
@Lukasa because I think it was wrong before. We would read EOF but we short-circuited it with something that isn't actually EOF.

Reply:
Ok. :)